docs/en/tutorial.md

```diff
@@ -341,7 +341,169 @@ Let's run some queries to analyze the 2M rows of data...
+  :::
+
+4. Verify it worked - you should get 265 rows (one row for each neighborhood):
+    ```sql
+    SELECT * FROM taxi_zone_dictionary
+    ```
+
+5. Use the `dictGet` function ([or its variations](./sql-reference/functions/ext-dict-functions.md)) to retrieve a value from a dictionary. You pass in the name of the dictionary, the value you want, and the key (which in our example is the `LocationID` column of `taxi_zone_dictionary`).
+
+    For example, the following query returns the `Borough` whose `LocationID` is 132 (which as we saw above is JFK airport):
+    ```sql
+    SELECT dictGet('taxi_zone_dictionary', 'Borough', 132)
+    ```
+
+    JFK is in Queens, and notice the time to retrieve the value is essentially 0:
+    ```response
+    ┌─dictGet('taxi_zone_dictionary', 'Borough', 132)─┐
+    │ Queens                                          │
+    └─────────────────────────────────────────────────┘
+
+    1 rows in set. Elapsed: 0.004 sec.
+    ```
+
+6. Use the `dictHas` function to see if a key is present in the dictionary. For example, the following query returns 1 (which is "true" in ClickHouse):
+    ```sql
+    SELECT dictHas('taxi_zone_dictionary', 132)
+    ```
+
+7. The following query returns 0 because 4567 is not a value of `LocationID` in the dictionary:
+    ```sql
+    SELECT dictHas('taxi_zone_dictionary', 4567)
+    ```
+
+8. Use the `dictGet` function to retrieve a borough's name in a query. For example:
+    ```sql
+    SELECT
+        count(1) AS total,
+        dictGetOrDefault('taxi_zone_dictionary','Borough', toUInt64(pickup_nyct2010_gid), 'Unknown') AS borough_name
+    FROM trips
+    WHERE dropoff_nyct2010_gid = 132 OR dropoff_nyct2010_gid = 138
+    GROUP BY borough_name
+    ORDER BY total DESC
+    ```
+
+    This query sums up the number of taxi rides per borough that end at either the LaGuardia or JFK airport. The result looks like the following, and notice there are quite a few trips where the dropoff neighborhood is unknown:
+    ```response
+    ┌─total─┬─borough_name──┐
+    │ 23683 │ Unknown       │
+    │  7053 │ Manhattan     │
+    │  6828 │ Brooklyn      │
+    │  4458 │ Queens        │
+    │  2670 │ Bronx         │
+    │   554 │ Staten Island │
+    │    53 │ EWR           │
+    └───────┴───────────────┘
+
+    7 rows in set. Elapsed: 0.019 sec. Processed 2.00 million rows, 4.00 MB (105.70 million rows/s., 211.40 MB/s.)
+    ```
+
+
+## 5. Perform a Join
+
+Let's write some queries that join the `taxi_zone_dictionary` with your `trips` table.
+
+1. We can start with a simple JOIN that acts similarly to the previous airport query above:
+    ```sql
+    SELECT
+        count(1) AS total,
+        Borough
+    FROM trips
+    JOIN taxi_zone_dictionary ON toUInt64(trips.pickup_nyct2010_gid) = taxi_zone_dictionary.LocationID
```

Comment:
Maybe add another queries to compare the speed? For customer to understand why he might need dictionaries? https://pastila.nl/?0007b917/37e107b6bf13dcd9507cb89e2da6faf4  There are might be some type conversion problems, but for quick demo it is Ok I think